pkg/combustion/kubernetes.go

```diff
@@ -105,16 +105,50 @@ func configureK3S(ctx *image.Context) (string, error) {
 		return "", fmt.Errorf("copying k3s installer script: %w", err)
 	}
 
+	configDir := generateComponentPath(ctx, k8sDir)
+	configPath := filepath.Join(configDir, k8sConfigDir)
+
+	cluster, err := kubernetes.NewCluster(&ctx.ImageDefinition.Kubernetes, configPath)
+	if err != nil {
+		return "", fmt.Errorf("initialising kubernetes cluster config: %w", err)
+	}
+
+	if err = storeKubernetesClusterConfig(cluster, ctx.CombustionDir); err != nil {
+		return "", fmt.Errorf("storing k3s cluster config: %w", err)
+	}
+
 	binaryPath, imagesPath, err := downloadK3sArtefacts(ctx)
 	if err != nil {
 		return "", fmt.Errorf("downloading k3s artefacts: %w", err)
 	}
 
 	templateValues := map[string]any{
+		"apiVIP":     ctx.ImageDefinition.Kubernetes.Network.APIVIP,
+		"apiHost":    ctx.ImageDefinition.Kubernetes.Network.APIHost,
 		"binaryPath": binaryPath,
 		"imagesPath": imagesPath,
 	}
 
+	singleNode := len(ctx.ImageDefinition.Kubernetes.Nodes) < 2
+	if singleNode {
+		var vipManifest string
+
+		if ctx.ImageDefinition.Kubernetes.Network.APIVIP == "" {
+			zap.S().Info("Virtual IP address for k3s cluster is not provided and will not be configured")
+		} else {
+			log.Audit("WARNING: A Virtual IP address for the k3s cluster has been provided. " +
+				"An external IP address for the Ingress controller (Traefik) must be manually configured.")
+			zap.S().Warn("Virtual IP address for k3s cluster is requested and will invalidate Traefik configuration")
```

Comment:
Proceeding with a warning is what we agreed on with @rdoxenham. In the future we'll extend the definition file with an `ingressIP` or something similar and automate this.

Reply:
Yep, that works.